### PR TITLE
Memory rework, xr_new to new

### PR DESCRIPTION
**Pull request by jazzvaz**
Needs testing

### CODE REVIEW

**Comment by jazzvaz** [comment]
It is better to do something with utility because they repeat engine files and each change should be also made there...

File: code/engine/xrCore/xrMemory.h [hunk 1]
```diff
@@ -1,85 +1,59 @@
 #pragma once
 
-class XRCORE_API xrMemory {
+class XRCORE_API xrMemory
+{
```
Comment (jazzvaz) [inline code/engine/xrCore/xrMemory.h]: format fixed in next commit

**Comment by Im-dex** [comment]
@jazzvaz 
>It is better to do something with utility because they repeat engine files and each change should be also made there...

Yep, they need some refactoring

**Comment by Im-dex** [comment]
@jazzvaz Works well to me

**Comment by jazzvaz** [comment]
@Im-dex I also have no problem

**Comment by Im-dex** [comment]
@jazzvaz Nice, just fix formatting and I will merge it

**Comment by jazzvaz** [comment]
@Im-dex which formatting? xr_memory format is fixed in the commit 

> replace xr_new with new

File: code/engine/xrCore/xrMemory.h (hunk 1 above)
Comment (Im-dex) [reply code/engine/xrCore/xrMemory.h]: still broken

File: code/engine/xrCore/xrMemory.h [hunk 2]
```diff
@@ -1,85 +1,48 @@
...
 
-template <bool _is_pm, typename T>
-struct xr_special_free {
-    IC void operator()(T*& ptr) {
-        void* _real_ptr = dynamic_cast<void*>(ptr);
-        ptr->~T();
-        Memory.mem_free(_real_ptr);
-    }
-};
+//#pragma warning(push)
```
Comment (Im-dex) [inline code/engine/xrCore/xrMemory.h]: remove this

**Review by Im-dex — CHANGES_REQUESTED** [review]

**Comment by jazzvaz** [comment]
@Im-dex can you test again, as I changed xrCDB memory calls.

**Comment by Im-dex** [comment]
@jazzvaz seems works nice, thanks.

**Pull request merged**